### PR TITLE
layers:Add binding for renderpass in pipeline

### LINKED ISSUE #91: Unify validation INVALID cmd buffer cases
Currently cmd buffers can be made invalid by updating or freeing descriptor sets that they depend on or if a framebuffer they reference is deleted.
Really, anytime that an object that a cmd buffer depends on is destroyed, the cmd buffer should be made invalid or, if it's already in-flight then an error should be flagged at destroy time.
Then when/if an INVALID cmd buffer is reported, an error should be flagged with a nice message about why the cmd buffer is invalid.

### PR DESCRIPTION
**Pull request by tobine**
Fixes #91 

When creating the cmd buffer<->object linkage, need to account for any
renderPass that is tied to the pipeline and bind it as well.

Added related test.

### CODE REVIEW

File: layers/core_validation.cpp [hunk 1]
```diff
@@ -7292,35 +7292,44 @@ ResetCommandBuffer(VkCommandBuffer commandBuffer, VkCommandBufferResetFlags flag
...
-                                 "Attempt to bind Pipeline 0x%" PRIxLEAST64 " that doesn't exist!", (uint64_t)(pipeline));
+            skip |= log_msg(dev_data->report_data, VK_DEBUG_REPORT_ERROR_BIT_EXT, VK_DEBUG_REPORT_OBJECT_TYPE_PIPELINE_EXT,
+                            (uint64_t)pipeline, __LINE__, DRAWSTATE_INVALID_PIPELINE, "DS",
+                            "Attempt to bind Pipeline 0x%" PRIxLEAST64 " that doesn't exist!", (uint64_t)(pipeline));
+        }
+        addCommandBufferBinding(&pipe_state->cb_bindings,
+                                {reinterpret_cast<uint64_t &>(pipeline), VK_DEBUG_REPORT_OBJECT_TYPE_PIPELINE_EXT}, cb_state);
+        if (VK_PIPELINE_BIND_POINT_GRAPHICS == pipelineBindPoint) {
+            // Add binding for child renderpass
+            auto rp_state = getRenderPassState(dev_data, pipe_state->graphicsPipelineCI.renderPass);
```
Comment (chrisforbes) [inline layers/core_validation.cpp]: :+1:

File: layers/core_validation.cpp [hunk 2]
```diff
@@ -7292,35 +7292,44 @@ ResetCommandBuffer(VkCommandBuffer commandBuffer, VkCommandBufferResetFlags flag
...
     std::unique_lock<std::mutex> lock(global_lock);
-    GLOBAL_CB_NODE *pCB = getCBNode(dev_data, commandBuffer);
-    if (pCB) {
-        skip_call |= addCmd(dev_data, pCB, CMD_BINDPIPELINE, "vkCmdBindPipeline()");
-        if ((VK_PIPELINE_BIND_POINT_COMPUTE == pipelineBindPoint) && (pCB->activeRenderPass)) {
-            skip_call |=
+    GLOBAL_CB_NODE *cb_state = getCBNode(dev_data, commandBuffer);
+    if (cb_state) {
+        skip |= addCmd(dev_data, cb_state, CMD_BINDPIPELINE, "vkCmdBindPipeline()");
+        if ((VK_PIPELINE_BIND_POINT_COMPUTE == pipelineBindPoint) && (cb_state->activeRenderPass)) {
```
Comment (chrisforbes) [inline layers/core_validation.cpp]: What a tangled mess of checks & side effects this function is... change isn't really making it any worse, though

File: layers/core_validation.cpp [hunk 3]
```diff
@@ -7292,35 +7292,44 @@ ResetCommandBuffer(VkCommandBuffer commandBuffer, VkCommandBufferResetFlags flag
...
+            set_pipeline_state(pipe_state);
         } else {
-            skip_call |= log_msg(dev_data->report_data, VK_DEBUG_REPORT_ERROR_BIT_EXT, VK_DEBUG_REPORT_OBJECT_TYPE_PIPELINE_EXT,
-                                 (uint64_t)pipeline, __LINE__, DRAWSTATE_INVALID_PIPELINE, "DS",
-                                 "Attempt to bind Pipeline 0x%" PRIxLEAST64 " that doesn't exist!", (uint64_t)(pipeline));
+            skip |= log_msg(dev_data->report_data, VK_DEBUG_REPORT_ERROR_BIT_EXT, VK_DEBUG_REPORT_OBJECT_TYPE_PIPELINE_EXT,
+                            (uint64_t)pipeline, __LINE__, DRAWSTATE_INVALID_PIPELINE, "DS",
+                            "Attempt to bind Pipeline 0x%" PRIxLEAST64 " that doesn't exist!", (uint64_t)(pipeline));
+        }
+        addCommandBufferBinding(&pipe_state->cb_bindings,
```
Comment (chrisforbes) [inline layers/core_validation.cpp]: Null ptr deref on `pipe_state` if the else branch was taken above. Not new, but more obvious now `getPipelineState`'s result is reused.

**Comment by tobine** [comment]
cherry-picked and pushed to master

**Pull request closed without merging**